packages/elements/list/src/insertFragmentList.ts

```diff
@@ -49,35 +73,134 @@ export const insertFragmentList = (editor: PlateEditor) => {
           ? commonAncestor
           : Node.common(listRoot, textEntry[1], commonAncestor[1]),
       // any list item would do, we grab the first one
-      getFirstAncestorOfType(listRoot, textEntries[0], li)
+      getFirstAncestorOfType(listRoot, textEntries[0], listItemPlugin)
     );
 
-    return isListRoot(commonAncestorEntry[0])
+    const [first, ...rest] = isListRoot(commonAncestorEntry[0])
       ? commonAncestorEntry[0].children
       : [commonAncestorEntry[0]];
+    return [...findListItemsWithContent(first), ...rest];
+  };
+
+  const wrapNodeIntoListItem = (node: TDescendant): TDescendant => {
+    return node.type === listItemType
+      ? node
+      : {
+          type: listItemType,
+          children: [node],
+        };
+  };
+
+  /**
+   * Checks if the fragment only consists of a single LIC in which case it is considered the user's intention was to copy a text, not a list
+   */
+  const isSingleLic = (fragment: TDescendant[]) => {
+    if (fragment.length !== 1 || !isListRoot(fragment[0])) return false;
+
+    const root = fragment[0];
+    // start at root path
+    const queue: Path[] = [[]];
+    let foundLic = false;
+    while (queue.length) {
+      const path = queue.shift()!;
+      const node = Node.get(root, path);
+      if (!Element.isElement(node)) {
+        return false;
+      }
+
+      if (isList(node)) {
+        const children = Array.from(Node.children(root, path)).map(
+          ([, p]) => p
+        );
+        queue.push(...children);
+      } else if ((node as TDescendant).type !== listItemContentType) {
+        return false;
+      } else if (!foundLic) {
+        foundLic = true;
+      } else {
+        return false;
+      }
+    }
+
+    return true;
   };
 
   return (fragment: TDescendant[]) => {
     const liEntry = findNode(editor, {
-      match: { type: li.type },
+      match: { type: listItemType },
       mode: 'lowest',
     });
+    // not inserting into a list item, delegate to other plugins
+    if (!liEntry) {
+      return insertFragment(
+        isListRoot(fragment[0]) ? [{ text: '' }, ...fragment] : fragment
+      );
+    }
 
-    if (liEntry) {
-      const [, liPath] = liEntry;
+    // delete selection (if necessary) so that it can check if needs to insert into an empty block
+    Transforms.insertFragment(editor, [{ text: '' }]);
```

Comment:
It is possible that the selection spans multiple LIs so there is no way to know at this point if the insert will happen inside an empty LIC. We need to check if the paste is to an empty block in order to decide if we want to update the content of the current block instead of inserting it as a new LI and leaving the empty LI.